packages/ui-kit-components/src/components/Button/button.test.tsx

```diff
@@ -1,11 +1,66 @@
 import '@testing-library/jest-dom';
 
-import { render, screen } from '@testing-library/react';
+import { act, fireEvent, render, screen } from '@testing-library/react';
 
 import Button from './button';
 
-test('Render button component correctly', () => {
-  render(<Button>Test Button</Button>);
+describe('Button component tests', () => {
+  it('Render button component correctly', () => {
+    render(<Button>Test Button</Button>);
 
-  expect(screen.getAllByText('Test Button')[0]).toBeInTheDocument();
+    expect(screen.getAllByText('Test Button')[0]).toBeInTheDocument();
+  });
+
+  it('Click button and the corresponding function will be called', () => {
+    const fn = jest.fn(() => {});
+
+    render(<Button click={fn}>Test Button</Button>);
+
+    fireEvent.click(screen.getByText('Test Button'));
+
+    expect(fn).toBeCalled();
+  });
```

Comment:
Generally it's better for each test to separate the setting and the testing:
- you can then have multiple tests for the same configuration of component
- you can have one `expect` per `it` block so you're testing one thing per test

Also since this is the beginning of the tests, you guys could setup some guidelines as to how to name the tests.
Could consider following the given/when/then pattern

example
```
describe('Given render button component', () => {

 let fn =  jest.fn(() => {});
 beforeEach(() => {
    render(<Button click={fn}>Test Button</Button>);
  })

  describe('When I click on button', () => {
    beforeEach(() => {
      fireEvent.click(screen.getByText('Test Button'));
    })
 
    it('Then the corresponding function should fire', ()=> {
       expect(fn).toBeCalled();
    })

  })
})

```